File: python_scripts/Cluster.py
class Cluster:
    jaccards = {}
    minHashes = []
    clusterId = ""
    clusterName = ""

    def __init__(self, clusterId):
        self.clusterId = clusterId

    def addMinhashes(self, minHashes):
        self.minHashes = minHashes

    def addName(self, name):
        self.clusterName = name

    def getMinhash(self):
        return self.minHashes

File: similarfiles/src/main/scala/com/edu/cloud/Cluster.scala


package com.edu.cloud

class Cluster(id: Int, fingerPrints: List[Long], jaccards: Map[Int, Double], minHashes: List[Long]) {
  var clusterId : Int = id;
  var fingerPrint: List[Long] = fingerPrints;
  var jaccard : Map[Int, Double] = jaccards
  var minHash : List[Long] = minHashes

  def getFingerPrint(): List[Long] = {
    return fingerPrint;
  }


  def setJaccard(j : Map[Int, Double]) = {
      jaccard = j;
  }
}


